std/btree.asm

```diff
@@ -0,0 +1,153 @@
+use std::utils::Option;
+
+/// This is a BTree, which means that each node has
+/// up to `max_items` items and a number of children
+/// that is one more than the number of items.
+/// The items are key-value-pairs sorted by key and
+/// the key of the i-th item is greater than all keys
+/// in children[i] and less than all keys in children[i + 1].
+/// Leaf nodes do not have children.
+enum BTree<K, V> {
+    Inner((K, V)[], BTree<K, V>[]),
+    Leaf((K, V)[]),
+}
+
+// Rust actually uses 11, but I have the impression that longer
+// arrays do not have that much of an advantage here.
+let max_items = 5;
```

Comment:
I think people choose odd numbers because they extract the middle value before the new value is added, for example here: https://en.wikipedia.org/wiki/2%E2%80%933%E2%80%934_tree

But in the end, it does not really matter.